lib/CMakeLists.txt

```diff
@@ -507,276 +110,47 @@ if(IS_MACOSX OR IS_LINUX)
-# macro expects the library name to be prefixed with _, but adds an underscore
-# to the package name when creating the Python extension module initialization
-# function (see Pixar's pyModule.h).
-if(IS_WINDOWS AND MAYAUSD_DEFINE_BOOST_DEBUG_PYTHON_FLAG)
-    # On Windows when compiling with debug python the library must be named with _d.
-    set(PYTHON_LIBRARY_NAME _${LIBRARY_NAME}_d)
-else()
-    set(PYTHON_LIBRARY_NAME _${LIBRARY_NAME})
-endif()
-
-add_library(${PYTHON_LIBRARY_NAME}
-    SHARED
-    python/module.cpp
-    python/wrapAdaptor.cpp
-    python/wrapBlockSceneModificationContext.cpp
-    python/wrapColorSpace.cpp
-    python/wrapDiagnosticDelegate.cpp
-    python/wrapMeshUtil.cpp
-    python/wrapQuery.cpp
-    python/wrapReadUtil.cpp
-    python/wrapRoundTripUtil.cpp
-    python/wrapStageCache.cpp
-    python/wrapUserTaggedAttribute.cpp
-    python/wrapWriteUtil.cpp
-    python/wrapXformStack.cpp
-)
-
-if(IS_MACOSX)
-    set(_macDef OSMac_)
-endif()
-
-target_compile_definitions(${PYTHON_LIBRARY_NAME}
-    PRIVATE
-        ${_macDef}
-        MFB_PACKAGE_NAME=${LIBRARY_NAME}
-        MFB_ALT_PACKAGE_NAME=${LIBRARY_NAME}
-        MFB_PACKAGE_MODULE=${LIBRARY_NAME}
-)
-set_target_properties(${PYTHON_LIBRARY_NAME}
-    PROPERTIES
-        PREFIX ""
-)
-if(IS_WINDOWS)
-    set_target_properties(${PYTHON_LIBRARY_NAME}
-        PROPERTIES
-            SUFFIX ".pyd"
-    )
-elseif(IS_MACOSX)
-    set_target_properties(${PYTHON_LIBRARY_NAME}
-        PROPERTIES
-            SUFFIX ".so"
-    )
-endif()
-
-target_link_libraries(${PYTHON_LIBRARY_NAME}
-    ${LIBRARY_NAME}
-)
-
-# We will use the following Python package structure:
-#
-# mayaUsd: top-level package
-# mayaUsd.lib: Python wrappers for mayaUsd library functionality.
-#
-# We re-map the Python library namespace to mayaUsd.lib through a simple
-# __init__.py.
-
-set(PYLIB_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME}/lib)
-
-if(IS_MACOSX OR IS_LINUX)
-    mayaUsd_init_rpath(rpath "lib")
-    mayaUsd_add_rpath(rpath "../../..")
-    if(IS_MACOSX AND DEFINED MAYAUSD_TO_USD_RELATIVE_PATH)
-        mayaUsd_add_rpath(rpath "../../../../../../Maya.app/Contents/MacOS")
-        mayaUsd_add_rpath(rpath "../../../../${MAYAUSD_TO_USD_RELATIVE_PATH}/lib")
-    endif()
-    mayaUsd_install_rpath(rpath ${PYTHON_LIBRARY_NAME})
-endif()
-
-install(TARGETS ${PYTHON_LIBRARY_NAME}
-    LIBRARY
-    DESTINATION ${PYLIB_INSTALL_PREFIX}
-    ARCHIVE
-    DESTINATION ${PYLIB_INSTALL_PREFIX}
-    RUNTIME
-    DESTINATION ${PYLIB_INSTALL_PREFIX}
-)
-if(IS_WINDOWS)
-    install(FILES $<TARGET_PDB_FILE:${PYTHON_LIBRARY_NAME}> DESTINATION ${PYLIB_INSTALL_PREFIX} OPTIONAL)
+    add_subdirectory(ufe)
 endif()
-
-install(FILES ae/__init__.py DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python/${PROJECT_NAME})
-install(FILES python/__init__.py DESTINATION ${PYLIB_INSTALL_PREFIX})
-
 add_subdirectory(usd)
+add_subdirectory(utils)
```

Comment:
One of the main problems that this PR addresses is to clean up this CMake file. 

- Every sub-directories under `lib` now gets it's own CMakeLists.txt

- All the `target_xxx` calls **explicitly** declare properties such as `PUBLIC`, `PRIVATE`. This reduces the chance to  introduce hidden dependencies.